src/coreclr/jit/hwintrinsic.cpp

```diff
@@ -1396,6 +1396,60 @@ GenTree* Compiler::impHWIntrinsic(NamedIntrinsic        intrinsic,
         GenTree* op3 = nullptr;
         GenTree* op4 = nullptr;
 
+        switch (numArgs)
+        {
+            case 4:
+                op4 = getArgForHWIntrinsic(sigReader.GetOp4Type(), sigReader.op4ClsHnd);
+                op4 = addRangeCheckIfNeeded(intrinsic, op4, mustExpand, immLowerBound, immUpperBound);
+                op3 = getArgForHWIntrinsic(sigReader.GetOp3Type(), sigReader.op3ClsHnd);
+                op2 = getArgForHWIntrinsic(sigReader.GetOp2Type(), sigReader.op2ClsHnd);
+                op1 = getArgForHWIntrinsic(sigReader.GetOp1Type(), sigReader.op1ClsHnd);
+                break;
+
+            case 3:
+                op3 = getArgForHWIntrinsic(sigReader.GetOp3Type(), sigReader.op3ClsHnd);
+                op2 = getArgForHWIntrinsic(sigReader.GetOp2Type(), sigReader.op2ClsHnd);
+                op1 = getArgForHWIntrinsic(sigReader.GetOp1Type(), sigReader.op1ClsHnd);
+                break;
+
+            case 2:
+                op2 = getArgForHWIntrinsic(sigReader.GetOp2Type(), sigReader.op2ClsHnd);
+                op2 = addRangeCheckIfNeeded(intrinsic, op2, mustExpand, immLowerBound, immUpperBound);
+                op1 = getArgForHWIntrinsic(sigReader.GetOp1Type(), sigReader.op1ClsHnd);
+                break;
+
+            case 1:
+                op1 = getArgForHWIntrinsic(sigReader.GetOp1Type(), sigReader.op1ClsHnd);
+                break;
+
+            default:
+                break;
+        }
+
+#if defined(TARGET_ARM64)
+        // Embedded masks need inserting as op1.
+        if (HWIntrinsicInfo::IsEmbeddedMaskedOperation(intrinsic))
+        {
+            numArgs++;
+            assert(numArgs <= 4);
+            switch (numArgs)
+            {
+                case 4:
+                    op4 = op3;
+                    FALLTHROUGH;
+                case 3:
+                    op3 = op2;
+                    FALLTHROUGH;
+                case 2:
+                    op2 = op1;
+                    FALLTHROUGH;
+                default:
+                    break;
+            }
+            op1 = gtNewSimdEmbeddedMaskNode(simdBaseJitType, simdSize);
```

Comment:
Why does this need to be done here?

This seems like its just inserting the implicit `AllTrue` mask that some instructions require, which is effectively allocating and forcing an extra node to be carried through all of HIR.

Seemingly this could just be inserted as part of lowering instead so that it only impacts LSRA and codegen?